### PR TITLE
Update Torch Elastic documentation

### LINKED ISSUE #8243: Update torchelstic docs in Lightning
TorchElastic is now upstream, part of torch.distributed.
We need to update [our docs](https://pytorch-lightning.readthedocs.io/en/stable/advanced/multi_gpu.html?highlight=elastic#torchelastic) with the new comands and point to their [new docs](https://pytorch.org/docs/stable/distributed.elastic.html).

They plan on keeping support for etcd, but it is recommended to use the native c10d instead

cc @cbalioglu

### PR DESCRIPTION
**Pull request by kaushikb11**
## What does this PR do?

TorchElastic is now upstream, part of torch.distributed.

Fixes #8243 

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [] Did you write any new necessary tests? (not for typos and docs)
- [] Did you verify new and existing tests pass locally with your changes?
- [] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by edenlightning — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]
eggsellent!

**Review by ananthsub — APPROVED** [review]
thanks @kaushikb11 !

File: docs/source/advanced/multi_gpu.rst [hunk 1]
```diff
@@ -639,38 +639,39 @@ The reason is that the full batch is visible to all GPUs on the node when using
...
+-------------------------
+Lightning supports the use of Torch Distributed Elastic to enable fault-tolerant and elastic distributed job scheduling. To use it, specify the 'ddp' or 'ddp2' backend and the number of gpus you want to use in the trainer.
 
 .. code-block:: python
 
     Trainer(gpus=8, accelerator='ddp')
 
-
-Following the `TorchElastic Quickstart documentation <https://pytorch.org/elastic/latest/quickstart.html>`_, you then need to start a single-node etcd server on one of the hosts:
+To launch a fault-tolerant job, run the following on all nodes.
```
Comment (ananthsub) [inline docs/source/advanced/multi_gpu.rst:650]: @edenafek - do you think its worth having a dedicated docs section on which schedulers/launchers Lightning supports? This could cover launching with SLURM, torch distributed elastic, etc

File: docs/source/advanced/multi_gpu.rst [hunk 2]
```diff
@@ -639,38 +639,39 @@ The reason is that the full batch is visible to all GPUs on the node when using
...
-    etcd --enable-v2
-         --listen-client-urls http://0.0.0.0:2379,http://127.0.0.1:4001
-         --advertise-client-urls PUBLIC_HOSTNAME:2379
-
+    python -m torch.distributed.run
+            --nnodes=NUM_NODES
+            --nproc_per_node=TRAINERS_PER_NODE
+            --rdzv_id=JOB_ID
+            --rdzv_backend=c10d
+            --rdzv_endpoint=HOST_NODE_ADDR
```
Comment (Borda) [inline docs/source/advanced/multi_gpu.rst:659]:
```suggestion
    python -m torch.distributed.run
            --nnodes=NUM_NODES
            --nproc_per_node=TRAINERS_PER_NODE
            --rdzv_id=JOB_ID
            --rdzv_backend=c10d
            --rdzv_endpoint=HOST_NODE_ADDR
```
can we add some samples how these variables shall look like? is it a string?

File: docs/source/advanced/multi_gpu.rst [hunk 3]
```diff
@@ -639,38 +639,39 @@ The reason is that the full batch is visible to all GPUs on the node when using
...
-Following the `TorchElastic Quickstart documentation <https://pytorch.org/elastic/latest/quickstart.html>`_, you then need to start a single-node etcd server on one of the hosts:
+To launch a fault-tolerant job, run the following on all nodes.
 
 .. code-block:: bash
 
-    etcd --enable-v2
-         --listen-client-urls http://0.0.0.0:2379,http://127.0.0.1:4001
-         --advertise-client-urls PUBLIC_HOSTNAME:2379
-
+    python -m torch.distributed.run
```
Comment (carmocca) [inline docs/source/advanced/multi_gpu.rst:654]: Doesn't these need backslashes?

File: docs/source/advanced/multi_gpu.rst [hunk 4]
```diff
@@ -639,38 +639,39 @@ The reason is that the full batch is visible to all GPUs on the node when using
...
+            --nnodes=NUM_NODES
+            --nproc_per_node=TRAINERS_PER_NODE
+            --rdzv_id=JOB_ID
+            --rdzv_backend=c10d
+            --rdzv_endpoint=HOST_NODE_ADDR
+            YOUR_LIGHTNING_TRAINING_SCRIPT.py (--arg1 ... train script args...)
 
-And then launch the elastic job with:
+To launch an elastic job, run the following on at least MIN_SIZE nodes and at most MAX_SIZE nodes.
```
Comment (carmocca) [inline docs/source/advanced/multi_gpu.rst:662]:
```suggestion
To launch an elastic job, run the following on at least ``MIN_SIZE`` nodes and at most ``MAX_SIZE`` nodes.
```

File: docs/source/advanced/multi_gpu.rst (hunk 2 above)
Comment (kaushikb11) [reply docs/source/advanced/multi_gpu.rst:659]: I don't think it's required here. Also, it's more like a reference.

File: docs/source/advanced/multi_gpu.rst (hunk 3 above)
Comment (kaushikb11) [reply docs/source/advanced/multi_gpu.rst:654]: I don't think it's required here. Also, it's more like a reference.

File: docs/source/advanced/multi_gpu.rst (hunk 2 above)
Comment (kaushikb11) [reply docs/source/advanced/multi_gpu.rst:659]: It would be advisable to learn more about it in the official Elastic docs in much more detail.
